src/main/java/seedu/address/model/filter/FilterByDate.java

```diff
@@ -0,0 +1,95 @@
+package seedu.address.model.filter;
+
+import java.time.LocalDateTime;
+import java.util.stream.Stream;
+
+import seedu.address.model.task.DeadlineTask;
+import seedu.address.model.task.EventTask;
+import seedu.address.model.task.FloatingTask;
+
+public class FilterByDate implements TaskPredicate{
```

Comment:
This class name is quite confusing. And this class should have some heading descriptions.